src/mbgl/text/harfbuzz.cpp

```diff
@@ -0,0 +1,121 @@
+#include <mbgl/text/harfbuzz.hpp>
+
+#include <hb-ft.h>
+
+namespace mbgl {
+
+struct InternalHBLangInfo {
+    std::string fontFileName;
+    std::string language;
+    hb_script_t script;
+    hb_direction_t direction;
+};
+
+std::map<GlyphIDType, InternalHBLangInfo> HBShaper::internalInfos = {
+    {GlyphIDType::Khmer, {"khmer.ttf", "kh", HB_SCRIPT_KHMER, HB_DIRECTION_LTR}},
+    {GlyphIDType::Myanmar, {"Myanmar.ttf", "my", HB_SCRIPT_MYANMAR, HB_DIRECTION_LTR}},
+    {GlyphIDType::Devanagari, {"Devanagari.ttf", "hi", HB_SCRIPT_DEVANAGARI, HB_DIRECTION_LTR}},
```

Comment:
Language: Would it make sense to introduce a `text-language` style spec property? Because for example Devanagari can be used for Hindi and Marathi and the shaping results might be different.

Reply:
Good question here. Does you mean when the script is different even the font name is same the render result is different?